src/main/java/org/gridsuite/ds/server/service/parameters/impl/ParametersServiceImpl.java

```diff
@@ -157,6 +155,18 @@ public DynamicSimulationRunContext createRunContext(UUID networkUuid, String var
             throw new DynamicSimulationException(PROVIDER_NOT_FOUND, "Dynamic simulation provider not found: " + runContext.getProvider());
         }
 
+        // set mapping for run context
+        String mappingToUse = mapping;
+        if (mappingToUse == null) {
+            mappingToUse = runContext.getParameters().getMapping();
+        }
+        runContext.setMapping(mappingToUse);
+
+        // check mapping
+        if (runContext.getMapping() == null) {
+            throw new DynamicSimulationException(PROVIDER_NOT_FOUND, "Dynamic simulation provider not found: " + runContext.getProvider());
```

Comment:
You just created a MAPPING_NOT_PROVIDED error type but throw a PROVIDER_NOT_FOUND error instead when the mapping is not found. Same remark for the error message.

Reply:
DONE! Tks!